### PR TITLE
feat(trufflehog): add Trufflehog action.

### PR DESCRIPTION
**Pull request by CharlieSu**
Based on https://github.com/edplato/trufflehog-actions-scan

* Uses requirement.txt so that Dependabot can provide updates.
* Sources regexes from the latest list.
* Uses a newer version of Trufflehog.

### CODE REVIEW

**Review by marcind — APPROVED** [review]

**Review by apulgarin-isp — APPROVED** [review]
looks good to me!

File: trufflehog-actions-scan/entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env sh
+
+set -e # Abort script at first error
+
+args="--regex --entropy=False --max_depth=50" # Default trufflehog options
+
+if [ -n "${INPUT_SCANARGUMENTS}" ]; then
```
Comment (jasonladuke0311) [inline trufflehog-actions-scan/entrypoint.sh:7]: is this supposed to be `INPUTS_SCANARGUMENTS` instead of input? Not sure how the env var is set.

**Review by jasonladuke0311 — APPROVED** [review]
Assuming that var is correct

File: trufflehog-actions-scan/entrypoint.sh (hunk 1 above)
Comment (jasonladuke0311) [reply trufflehog-actions-scan/entrypoint.sh:7]: disregard

**Pull request merged**